analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -196,6 +199,25 @@ private static ProgramState[] ProcessEquals(SymbolicContext context, IInvocation
         return context.State.ToArray();
     }
 
+    private static ProgramState[] ProcessNullableGetValueOrDefault(SymbolicContext context, IInvocationOperationWrapper invocation)
+    {
+        var instanceValue = context.State[invocation.Instance];
+        if (instanceValue?.HasConstraint(ObjectConstraint.Null) is true)
+        {
+            return ConstantCheck.ConstraintFromType(((INamedTypeSymbol)invocation.Instance.Type).TypeArguments.Single()) is { } orDefaultConstraint
+                ? context.SetOperationConstraint(orDefaultConstraint).ToArray()
+                : context.State.ToArray();
+        }
+        else if (instanceValue is not null)
+        {
+            return context.State.SetOperationValue(invocation, instanceValue).ToArray();
```

Comment:
The description says
> NotNull => propagate constraints like with nullable.Value

but the NotNull check is missing here. Is it intentional to always copy as long as "Null" is not present?

```suggestion
        else if (instanceValue.HasConstraint(ObjectConstraint.NotNull))
        {
            return context.State.SetOperationValue(invocation, instanceValue).ToArray();
```

Reply:
It is equivalent. 
If it's `Null`, we went to `if` branch and didn't come here.
If it's `<null>`, we don't go in.
Otherwise, it must be `NotNull` because of nature of nullable value types. And even if it wouldn't be, it's fine just to propagate whatever constraints are there.

Reply:
> Otherwise, it must be NotNull 

That's the point. It also could be an empty SymbolicValue or a SymbolicValue with another constraint. Why don't we switch over the result of `context.State[invocation.Instance].HasConstraint<ObjectConstraint>()` instead? That would be much clearer (Null, NotNull, No ObjectContraint).

Reply:
In case it's empty, or anything with other constraints, we want to propagate the full `instanceValue` as it was.

Same as
```
nullable.Value
^^^^^^^^       all constraints from here (the whole Symbolic Value)
         ^^^^^ are propagated here
```
The `GetValueOrDefault()` should behave exactly the same for the `GetValue` part functionality